### PR TITLE
Add array module documentation

### PR DESCRIPTION
**Pull request by daviditen**
Finished the module header section for chpldocs for ChapelArray.chpl
and either commented or "no doc"'ed all functions that aren't filtered
out by default.

### CODE REVIEW

File: modules/internal/ChapelArray.chpl [hunk 1]
```diff
@@ -19,6 +19,118 @@
...
+
+   .. code-block:: chapel
+
+     A op= B;
+
+   ``A`` must not share its domain with another array, otherwise the program
+   will halt with an error message.
+
+   For the ``+=`` and ``|=`` operators, the value from ``B`` will overwrite
+   the existing value in ``A`` when indices overlap.
```
Comment (benharsh) [inline modules/internal/ChapelArray.chpl]: Might be useful to suggest that the user look at the sets readme for more information.

**Comment by benharsh** [comment]
Could it be worthwhile to link to the arrays and domains primers?

**Comment by benharsh** [comment]
Looks good to me! This could go in without those links, your choice.

**Comment by daviditen** [comment]
Thanks for reviewing Ben.

I think your comment about pointing to the sets README is a very good suggestion and I tried to add a cross-reference before remembering that there's a small issue with our doc builds when we do that.  I think I'd prefer to leave this out for now and add it after the doc builds can handle this cross-reference instead of trying to add it as a full path to the .rst or something like that.

If there was a single mega-primer for all arrays and domains I think I'd be for linking to it, but as it is currently there are a small pile of different primers that would need to be mentioned for reasonable coverage.  I'd like to leave this out for now too.

**Pull request merged**